### DISCUSSION TITLE
Can I match / ignore files in folders without knowing it's folder structure?

### DISCUSSION THREAD

**Discussion by ziontee113** [BurntSushi/ripgrep#2287] [Q&A] [answered]
Thank you so much for the amazing project. And I apologize before hand if I'm doing a bad job asking my question.

Let's say we have this folder structure:

```
├── lua
│   └── portals
│       ├── core.lua
│       ├── init.lua
│       ├── utils
│       │   └── fmt.lua
│       └── utils.lua
└── tests
    ├── core
    │   └── core_spec.lua
    ├── minimal_init.lua
    └── utils
        ├── above_and_below_spec.lua
        ├── buffer_spec.lua
        └── treesitter_spec.lua
```
I can do `rg "the_string_to_search" lua/` to search in `lua` folder only, or `rg "the_string_to_search" -g "!lua/"` to ignore that folder.

If I want to ignore `lua/potals/utils` folder, I'd have to do: `rg "the_string_to_search" -g "!lua/portals/utils"`
or `rg "the_string_to_search" -g "!*/*/utils"`

My question is: if I don't know beforehand the folder structure that leads to `utils` folder, or if the folder structure is too complex, I won't know how many `*/` to add to the command. Is there a way that I can achieve the same thing as `rg "the_string_to_search" -g "!*/*/utils` without knowing how many `*/` to add?

Thank you very much!

**Comment by ziontee113** [comment, marked as answer]
Hi, I think I've found the answer from:
https://github.com/BurntSushi/ripgrep/issues/831
https://docs.rs/globset/0.3.0/globset/#syntax

I tried `rg "the_string_to_search" -g "!**/utils/*.lua"` and it seemed to work :smile:

**Reply by BurntSushi** [reply]
Exactly! Nice find.
